src/server.js

```diff
@@ -298,19 +291,18 @@ class Server extends EventEmitter {
 
                 if (funcResult && funcResult.upstreamProxyUrl) {
                     try {
-                        handlerOpts.upstreamProxyUrlParsed = parseUrl(funcResult.upstreamProxyUrl);
+                        handlerOpts.upstreamProxyUrlParsed = new URL(funcResult.upstreamProxyUrl);
                     } catch (e) {
                         throw new Error(`Invalid "upstreamProxyUrl" provided: ${e} (was "${funcResult.upstreamProxyUrl}"`);
                     }
-
-                    if (!handlerOpts.upstreamProxyUrlParsed.hostname || !handlerOpts.upstreamProxyUrlParsed.port) {
-                        // eslint-disable-next-line max-len
-                        throw new Error(`Invalid "upstreamProxyUrl" provided: URL must have hostname and port (was "${funcResult.upstreamProxyUrl}")`);
-                    }
                     if (handlerOpts.upstreamProxyUrlParsed.protocol !== 'http:') {
                         // eslint-disable-next-line max-len
                         throw new Error(`Invalid "upstreamProxyUrl" provided: URL must have the "http" protocol (was "${funcResult.upstreamProxyUrl}")`);
                     }
+                    if (!handlerOpts.upstreamProxyUrlParsed.hostname) {
+                        // eslint-disable-next-line max-len
+                        throw new Error(`Invalid "upstreamProxyUrl" provided: URL must have hostname (was "${funcResult.upstreamProxyUrl}")`);
+                    }
```

Comment:
Why do we no longer expect port?

Reply:
We do. It's already covered by the protocol check. It's an empty string if it's a default port.

Reply:
Ok. I'm just pointing it out because previously we expected a port number here, so if we don't anymore, somewhere down the line we have to be sure that `''` will also work.